class/hd-segwit-p2sh-ar-wallet.js

```diff
@@ -49,17 +53,53 @@ export class HDSegwitP2SHArWallet extends AbstractHDSegwitP2SHWallet {
     return address;
   }
 
-  createTx({ utxos, amount, fee, address, privateKeys = [], vaultTxType }) {
+  async mnemonicToKeyPair(mnemonic) {
+    const SALT_LENGHT = 4;
+    const WORD_BIT_LENGHT = 11;
+    const WORDS_LENGTH = 12;
+
+    const words = mnemonic.split(' ');
+
+    if (words.length !== WORDS_LENGTH) {
+      throw new Error(`Provided ${words.length} words expected ${WORDS_LENGTH}`);
+    }
+
+    const bits132 = mnemonic.split(' ').reduce((bits, word) => {
+      const index = bip39.wordlists.english.indexOf(word);
+      if (index === -1) {
+        throw new Error(`Couldn't find index for word: ${word}`);
```

Comment:
It would be good to keep errors from the 'backend' more generic so that it would be easier to map them with proper readable translations.

Reply:
changed errors into translation keys